langserver/completion.go

```diff
@@ -130,20 +129,24 @@ func (s *server) completeMetricName(ctx context.Context, completions *[]protocol
 		return err
 	}
 
+	names = make([]string, len(queries))
+	i = 0
 	for _, q := range queries {
-		if rec := q.Record; rec != "" && strings.HasPrefix(rec, metricName) {
-			item := protocol.CompletionItem{
-				Label:            rec,
-				SortText:         "__2__" + rec,
-				Kind:             3, //Value
-				InsertTextFormat: 2, //Snippet
-				TextEdit: &protocol.TextEdit{
-					Range:   editRange,
-					NewText: rec,
-				},
-			}
-			*completions = append(*completions, item)
+		names[i] = q.Record
+		i++
+	}
+	for _, match := range fuzzy.Find(metricName, names) {
+		item := protocol.CompletionItem{
+			Label:            match.Str,
+			SortText:         fmt.Sprintf("__2__%d", match.Score),
```

Comment:
This doesn't seem right. Here an integer is converted to a string which is then sorted. But for strings we have `"2" > "12"`.

(I missed that when looking at the last PR).

Reply:
ah! so we could use fmt.Sprintf("__2__%s", match.Str) instead?

Reply:
We do want to sort by match quality, not alphabetically.